search/intermediate_result.cpp

```diff
@@ -168,40 +161,60 @@ string PreResult2::GetRegionName(storage::CountryInfoGetter const & infoGetter,
   return info.m_name;
 }
 
+namespace
+{
+// TODO: Format street and house number according to local country's rules.
+string FormatStreetAndHouse(ReverseGeocoder::Address const & addr)
+{
+  ASSERT_GREATER_OR_EQUAL(addr.GetDistance(), 0, ());
+  return addr.GetStreetName() + ", " + addr.GetHouseNumber();
+}
+// TODO: Share common formatting code for search results and place page.
+string FormatFullAddress(ReverseGeocoder::Address const & addr, string const & region)
+{
+  // TODO: Print "near" for not exact addresses.
+  if (addr.GetDistance() != 0)
+    return region;
+
+  return FormatStreetAndHouse(addr) + (region.empty() ? "" : ", ") + region;
```

Comment:
Да, сделаю красиво, когда вынесу в пошаренный с другим кодом форматтер. Спасибо!

> On Mar 17, 2016, at 16:15, ygorshenin <notifications@github.com> wrote:
> 
> In search/intermediate_result.cpp <https://github.com/mapsme/omim/pull/2288#discussion_r56500958>:
> 
> > +namespace
> > +{
> > +// TODO: Format street and house number according to local country's rules.
> > +string FormatStreetAndHouse(ReverseGeocoder::Address const & addr)
> > +{
> > +  ASSERT_GREATER_OR_EQUAL(addr.GetDistance(), 0, ());
> > +  return addr.GetStreetName() + ", " + addr.GetHouseNumber();
> > +}
> > +// TODO: Share common formatting code for search results and place page.
> > +string FormatFullAddress(ReverseGeocoder::Address const & addr, string const & region)
> > +{
> > +  // TODO: Print "near" for not exact addresses.
> > +  if (addr.GetDistance() != 0)
> > +    return region;
> > +
> > +  return FormatStreetAndHouse(addr) + (region.empty() ? "" : ", ") + region;
> Мб не так хакерски?
> 
> string s = FormatStreetAndHouse(addr);
> if (!region.empty())
> {
>   s.append(", ");
>   s.append(region);
> }
> return s;
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2288/files/84cf9b0f444fb0b55f9b956001d32bbd92f008b3#r56500958>